### PR TITLE
Format docstrings for correct display in sphinx-generated bootstrap pages

### PR DESCRIPTION
**Pull request by santiagogaray**
Hi @chriswmackey. 
I did good progress but unfortunately I wasn't able to generate the documentation due to some errors. Looks like a similar case like last time where the libraries aren't up to date with this package. Is this similarly fixable by uninstalling some dependencies?
These are the error messages (sorry for the long post!):

C:\Users\sgara\Documents\GitHub\honeybee-energy\docs\cli.rst:4: WARNING: Failed to import "energy" from "honeybee_energy.cli". The following exception was raised:
Traceback (most recent call last):
  File "c:\python27\lib\site-packages\sphinx_click\ext.py", line 314, in _load_module
    mod = __import__(module_name, globals(), locals(), [attr_name])
  File "C:\Users\sgara\Documents\GitHub\honeybee-energy\honeybee_energy\__init__.py", line 8, in <module>
    import honeybee_energy._extend_honeybee
  File "C:\Users\sgara\Documents\GitHub\honeybee-energy\honeybee_energy\_extend_honeybee.py", line 19, in <module>
    from .properties.model import ModelEnergyProperties
  File "C:\Users\sgara\Documents\GitHub\honeybee-energy\honeybee_energy\properties\model.py", line 5, in <module>
    from ..lib.constructionsets import generic_construction_set
  File "C:\Users\sgara\Documents\GitHub\honeybee-energy\honeybee_energy\lib\constructionsets.py", line 2, in <module>
    from honeybee_energy.constructionset import ConstructionSet
  File "C:\Users\sgara\Documents\GitHub\honeybee-energy\honeybee_energy\constructionset.py", line 5, in <module>
    from .material.opaque import EnergyMaterial, EnergyMaterialNoMass
  File "C:\Users\sgara\Documents\GitHub\honeybee-energy\honeybee_energy\material\opaque.py", line 10, in <module>
    from ..writer import generate_idf_string
  File "C:\Users\sgara\Documents\GitHub\honeybee-energy\honeybee_energy\writer.py", line 6, in <module>
    from honeybee.facetype import RoofCeiling, AirBoundary
ImportError: cannot import name AirBoundary
WARNING: autodoc: failed to import module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'altnumber' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'boundarycondition' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'cli' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'cli.simulate' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'cli.translate' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'config' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'construction' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'construction.air' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'construction.opaque' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'construction.shade' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'construction.window' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'constructionset' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'hvac' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'hvac.idealair' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'lib' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'lib.constructions' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'lib.constructionsets' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'lib.data' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'lib.materials' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'lib.programtypes' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'lib.schedules' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'lib.scheduletypelimits' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'load' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'load.equipment' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'load.infiltration' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'load.lighting' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'load.people' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'load.setpoint' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'load.ventilation' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'material' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'material.gas' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'material.glazing' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'material.opaque' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'material.shade' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'programtype' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'properties' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'properties.aperture' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'properties.door' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'properties.face' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'properties.model' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'properties.room' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'properties.shade' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'reader' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'run' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'schedule' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'schedule.day' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'schedule.fixedinterval' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'schedule.rule' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'schedule.ruleset' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'schedule.typelimit' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'simulation' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'simulation.control' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'simulation.daylightsaving' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'simulation.output' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'simulation.parameter' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'simulation.runperiod' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'simulation.shadowcalculation' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'simulation.sizing' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
WARNING: autodoc: failed to import module u'writer' from module u'honeybee_energy'; the following exception was raised:
cannot import name AirBoundary
C:\Users\sgara\Documents\GitHub\honeybee-energy\docs\index.rst:2: WARNING: Title underline too short.

Welcome to Honeybee Energy's documentation!
===================================
docs/cli.rst:2: WARNING: Title underline too short.

honeybee energy command line interface
===============================
docs/cli.rst:2: WARNING: Title underline too short.

honeybee energy command line interface
===============================
docs/cli.rst:4: WARNING: Failed to import "energy" from "honeybee_energy.cli". The following exception was raised:
Traceback (most recent call last):
  File "c:\python27\lib\site-packages\sphinx_click\ext.py", line 314, in _load_module
    mod = __import__(module_name, globals(), locals(), [attr_name])
  File "C:\Users\sgara\Documents\GitHub\honeybee-energy\honeybee_energy\__init__.py", line 8, in <module>
    import honeybee_energy._extend_honeybee
  File "C:\Users\sgara\Documents\GitHub\honeybee-energy\honeybee_energy\_extend_honeybee.py", line 19, in <module>
    from .properties.model import ModelEnergyProperties
  File "C:\Users\sgara\Documents\GitHub\honeybee-energy\honeybee_energy\properties\model.py", line 5, in <module>
    from ..lib.constructionsets import generic_construction_set
  File "C:\Users\sgara\Documents\GitHub\honeybee-energy\honeybee_energy\lib\constructionsets.py", line 2, in <module>
    from honeybee_energy.constructionset import ConstructionSet
  File "C:\Users\sgara\Documents\GitHub\honeybee-energy\honeybee_energy\constructionset.py", line 5, in <module>
    from .material.opaque import EnergyMaterial, EnergyMaterialNoMass
  File "C:\Users\sgara\Documents\GitHub\honeybee-energy\honeybee_energy\material\opaque.py", line 10, in <module>
    from ..writer import generate_idf_string
  File "C:\Users\sgara\Documents\GitHub\honeybee-energy\honeybee_energy\writer.py", line 6, in <module>
    from honeybee.facetype import RoofCeiling, AirBoundary
ImportError: cannot import name AirBoundary

### CODE REVIEW

**Comment by chriswmackey** [comment]
Thanks for getting this together @santiagogaray . The following command should update your library so you can build the docs:
```
pip install honeybee-core -U
```
I'll look through your changes soon.

**Comment by santiagogaray** [comment]
Thanks @chriswmackey 
Unfortunately there are some errors while building the docs.
This time there are all of the type `SyntaxError: EOL while scanning string literal`
if you need more detail I can send you an email with it(is a lot longer than the last one.)

**Comment by chriswmackey** [comment]
@santiagogaray ,
I just cloned your fork locally and I was able to successfully build the docs but there are a lot of warnings. A number of them are `.. code-block:: json` which should be reformatted to `.. code-block:: python` among other warnings. None of them seem to be `SyntaxErrors`, though. Can you give me the module where the `SyntaxError` is coming from? That should make it clear which of your libraries need to be updated.

**Comment by santiagogaray** [comment]
Hi @chriswmackey 
The for wasn;t updated with my latest change, I just pushed the changed now as you can see.
Interesting that you didn't get any errors
See the txt file attached with all the messages.
[New Text Document.txt](https://github.com/ladybug-tools/honeybee-energy/files/4206167/New.Text.Document.txt)

File: honeybee_energy/programtype.py [hunk 1]
```diff
@@ -34,35 +57,11 @@ class ProgramType(object):
     """
     __slots__ = ('_name', '_people', '_lighting', '_electric_equipment',
                  '_gas_equipment', '_infiltration', '_ventilation',
-                 '_setpoint', '_locked')
+                 '_setpoint', '_locked)
```
Comment (chriswmackey) [inline honeybee_energy/programtype.py]: The reason why you get an error is because you accidentally deleted a `'` here.

**Comment by chriswmackey** [comment]
@santiagogaray . I pulled your changes and I can see the error. I put a comment for the  location of the error you should correct.

File: honeybee_energy/programtype.py (hunk 1 above)
Comment (santiagogaray) [reply honeybee_energy/programtype.py]: Thanks @chriswmackey for that, I guess I catched it once I realized that the mistake could be mine and not a dependency issue,,,sorry the extra work! 
Ironing out other bugs in the docs now...

**Comment by santiagogaray** [comment]
Hi @chriswmackey 
I think I fixed everything I could find but feel free to take a look when you get a chance. A couple of things though that may be minor:

I'm still getting one error, not sure what the issue in the rst file is, I wonder if you get the smae error on your side:
C:\Users\sgara\Documents\GitHub\honeybee-energy\docs\honeybee_energy.simulation.parameter.rst:6: WARNING: Could not lex literal_block as "python". Highlighting skipped.

Also I wonder if in this function we need too include the code-block with the dictionary:

image

**Comment by chriswmackey** [comment]
Thank you for getting this together @santiagogaray . I checked and the reason why you get that warning is because `//` is not correct Python syntax but we still have it within the code-block of the SimulationParameter class:
https://github.com/santiagogaray/honeybee-energy/blob/master/honeybee_energy/simulation/parameter.py#L276
These should be replaced with `#`.

I'll look through your other changes as soon as I get the chance.

**Comment by santiagogaray** [comment]
Thanks @chriswmackey 
I did fix quite a few instances where '//' was used. I guess it didn't look too out of place for me the javascript version.

Also I did notice the entire properties package does not include dictionary examples in the from_dict() function. Is this by design? Should I work on that?

**Comment by chriswmackey** [comment]
Thank you @santiagogaray . I'll look over everything today.

The sample dictionaries for the properties would be helpful. The reason why I didn't put them there in the first place was because the schema for these objects was changing a lot in in the early stages and there was a greater risk of the sample being misleading rather than helpful. But things are more stable now and so having samples there would be beneficial.

File: honeybee_energy/construction/window.py [hunk 2]
```diff
@@ -51,24 +51,25 @@ def materials(self):
...
+        * An exterior shade/screen/blind must be the first layer
+        * An interior shade/blind must be the last layer
+        * An interior screen is not allowed
+        * For an exterior shade/screen/blind or interior shade/blind, there should
+          not be a gas layer between the shade/screen/blind and adjacent glass
+          (we take care of this for shade materials)
+        * A between-glass screen is not allowed
+        * A between-glass shade/blind is allowed only for double and triple glazing
+        * A between-glass shade/blind must have adjacent gas layers of the same type
+          and width (we take care of this so the user does not specify the gaps)
```
Comment (chriswmackey) [inline honeybee_energy/construction/window.py]: This point is also meaningless now. Let's remove it.

File: honeybee_energy/construction/window.py [hunk 3]
```diff
@@ -51,24 +51,25 @@ def materials(self):
...
+        * For triple glazing the between-glass shade/blind must be between the two
+          inner glass layers. (currently no check)
+        * The slat width of a between-glass blind must be less than the sum of the
+          widths of the gas layers adjacent to the blind. (currently no check).
```
Comment (chriswmackey) [inline honeybee_energy/construction/window.py]: This should be changed to:
```
The slat width of a between-glass blind must be less than 2 times the blind's distance_to_glass.
(currently no check).
```

**Comment by chriswmackey** [comment]
Actually, the more that I think about it, I'd rather merge this in sooner than later. There's a refactoring that I need to do to one of the classes in this library, which will involve rewriting some of the docstrings on it and will create conflicts. I don't have to change anything on the classes that I commented on or the properties classes. So I say that we merge this and then we can easily work independently: I can work on refactoring that one class and you can work on addressing the comments in a separate PR. Then we can each send PRs that won't conflict with one another.

**Comment by santiagogaray** [comment]
Sounds good @chriswmackey 
I will create a future PR with your docstring revisions and dictionary samples. 
Thanks Chris again for timely helping me fix some of my mistakes!

**Pull request merged**